src/main/scala/lisa/kernel/fol/CommonDefinitions.scala

```diff
@@ -20,6 +20,12 @@ private[fol] trait CommonDefinitions {
     val id: String
   }
 
+
+  /**
+   * Marks classes that can represent symbols in a theory
+   */
+  trait TheorySymbol
```

Comment:
This is a suggestion, but perhaps we could have
```Scala
trait ConstantLabel
trait SchematicLabel // Possibly this one too
```
as theory symbols are always constant (that way we stick to the existing terminology). The front [does define `SchematicLabel`](https://github.com/FlorianCassayre/master-project/blob/master/src/main/scala/me/cassayre/florian/masterproject/front/fol/definitions/CommonDefinitions.scala#L14) because it helped factor out some of the logic, so it would be a sane addition.

Reply:
I think it makes sense to do it this way, by refining Label

Reply:
Sounds good to me, I hope it won't be confusing to have these 4 <Something>Label (otherwise the names are descriptive).